src/mcpdotnet/Utils/Json/JsonSerializerOptionsExtensions.cs

```diff
@@ -1,32 +1,72 @@
-﻿using System.Text.Json;
+﻿using McpDotNet.Protocol.Messages;
+using McpDotNet.Protocol.Types;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Text.Json.Nodes;
 using System.Text.Json.Serialization;
+using System.Text.Json.Serialization.Metadata;
 
 namespace McpDotNet.Utils.Json;
 
 /// <summary>
 /// Extensions for configuring System.Text.Json serialization options for MCP.
 /// </summary>
-internal static class JsonSerializerOptionsExtensions
+internal static partial class JsonSerializerOptionsExtensions
 {
     public static JsonSerializerOptions DefaultOptions { get; } = CreateDefaultOptions();
 
     /// <summary>
     /// Creates default options to use for MCP-related serialization.
     /// </summary>
     /// <returns>The configured options.</returns>
+    [UnconditionalSuppressMessage("AotAnalysis", "IL3050", Justification = "DefaultJsonTypeInfoResolver is only used when reflection-based serialization is enabled")]
+    [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026", Justification = "DefaultJsonTypeInfoResolver is only used when reflection-based serialization is enabled")]
     private static JsonSerializerOptions CreateDefaultOptions()
```

Comment:
Would it make sense to create this as chained with AIJsonUtilities.DefaultOptions?